crates/biome_js_analyze/src/lint/nursery/use_index_of/find_index_comparable_expression.rs

```diff
@@ -0,0 +1,78 @@
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsFunctionBody, JsAssignmentExpression, JsBinaryExpression,
+    JsLogicalExpression, JsReturnStatement, JsSyntaxNode, JsVariableDeclaration,
+};
+use biome_rowan::AstNode;
+
+fn node_is_undefined(node: &AnyJsExpression) -> bool {
+    node.syntax().text_trimmed() == "undefined"
+}
+
+fn extract_simple_compare_match(
+    expression: &JsBinaryExpression,
+    parameter_name: &String,
+) -> Option<JsSyntaxNode> {
+    if expression.operator_token().ok()?.text_trimmed() != "===" {
+        return None;
+    }
+
+    let (left, right) = (expression.left().unwrap(), expression.right().unwrap());
+
+    let matching_side = if left.to_trimmed_string() == *parameter_name {
+        right
+    } else if right.to_trimmed_string() == *parameter_name {
+        left
+    } else {
+        return None;
+    };
+
+    if node_is_undefined(&matching_side) {
+        return None;
+    }
+
+    Some(matching_side.syntax().clone())
```

Comment:
We could avoid the `clone()`:
```suggestion
    Some(matching_side.into_syntax())
```